test/new-e2e/tests/process/testing.go

```diff
@@ -155,6 +190,30 @@ func processDiscoveryHasData(disc *agentmodel.ProcessDiscovery) bool {
 	return disc.Pid != 0 && disc.Command.Ppid != 0 && len(disc.User.Name) > 0
 }
 
+// findContainer returns whether the container with the given image tag exists in the given list of
+// container payloads and whether it has the expected data populated
+func findContainer(image string, containers []*agentmodel.Container) (found, populated bool) {
+	for _, container := range containers {
+		for _, tag := range container.Tags {
+			if tag == image {
```

Comment:
optional: might be able to use [slices.Contains](https://pkg.go.dev/golang.org/x/exp/slices#Contains) to simplify this a little

Reply:
Done!